tests/ignite/metrics/test_epoch_metric.py

```diff
@@ -0,0 +1,93 @@
+from ignite.metrics import EpochMetric
+import torch
+import pytest
+
+
+def test_epoch_metric():
+
+    def compute_fn(y_preds, y_targets):
+        return 0.0
+
+    em = EpochMetric(compute_fn)
+
+    # Wrong input dims
+    with pytest.raises(AssertionError):
+        output = (torch.tensor(0), torch.tensor(0))
+        em.update(output)
+
+    # Wrong input dims
+    with pytest.raises(AssertionError):
+        output = (torch.rand(4, 3, 1), torch.rand(4, 3))
+        em.update(output)
+
+    # Target is not binary
+    with pytest.raises(AssertionError):
+        output = (torch.rand(4, 3), torch.randint(0, 5, size=(4, 3)))
+        em.update(output)
+
+    torch.manual_seed(12)
+    em.reset()
+    output1 = (torch.rand(4, 3), torch.randint(0, 2, size=(4, 3), dtype=torch.long))
+    em.update(output1)
+    output2 = (torch.rand(4, 3), torch.randint(0, 2, size=(4, 3), dtype=torch.long))
+    em.update(output2)
+
+    assert em._predictions.device.type == 'cpu' and em._targets.device.type == 'cpu'
+    assert torch.equal(em._predictions[:4, :], output1[0])
+    assert torch.equal(em._predictions[4:, :], output2[0])
+    assert torch.equal(em._targets[:4, :], output1[1])
+    assert torch.equal(em._targets[4:, :], output2[1])
+    assert em.compute() == 0.0
```

Comment:
might be better to make compute_fn a mock and assert its called with the expected tensors

Reply:
ah I see you have some examples  like this below. So this is not so important here

Reply:
okay, so i ignore this